bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/SVGFileFormat.java

```diff
@@ -31,7 +31,11 @@
 public class SVGFileFormat extends FileFormat {
 
 	/** The instance of the registered {@link SVGRasterizer}. */
-	private static final SVGRasterizer RASTERIZER = ServiceLoader.load(SVGRasterizer.class).findFirst().orElse(null);
+	// The service loader uses the classloader of the SVGRasterizer class to avoid
+	// that a potentially non-fitting context classloader is used for searching for
+	// a rasterizer implementation (such as during test execution).
+	private static final SVGRasterizer RASTERIZER = ServiceLoader
+			.load(SVGRasterizer.class, SVGRasterizer.class.getClassLoader()).findFirst().orElse(null);
```

Comment:
```suggestion
			.load(SVGRasterizer.class, SVGFileFormat.class.getClassLoader()).findFirst().orElse(null);
```

Even though `SVGRasterizer` and `SVGFileFormat` are in the same bundle and have the same classloader it seems semantically more correct to use `SVGFileFormat` as this is the class that loads the `SVGRasterizer` (what usually could be an API class from somewhere else).

Using an explicit classloader for serviceloader is best-practice anyways so I'm not sure we need an explicit comment here, but it of course does not harm.

Reply:
Independently which version of the patch is used, I can confirm the original test and the simplified one are working now on RHEL 9.2. Thanks.

Reply:
> Even though `SVGRasterizer` and `SVGFileFormat` are in the same bundle and have the same classloader it seems semantically more correct to use `SVGFileFormat` as this is the class that loads the `SVGRasterizer` (what usually could be an API class from somewhere else).

I was somehow concerned that it's a bad idea to use the class loader of the class that is currently being loaded (as we are in the static initialization. But since it works fine and I agree that it is semantically more correct, I adapted the PR accordingly.

> Using an explicit classloader for serviceloader is best-practice anyways so I'm not sure we need an explicit comment here, but it of course does not harm.

Thank you for the pointer. I wasn't aware of this. So I just removed the comment as it seems unnecessary to document best practice explicitly again.

> Independently which version of the patch is used, I can confirm the original test and the simplified one are working now on RHEL 9.2. Thanks.

Thank you for testing! Good to know that it works for you. I have also successfuly retested the modified change on Windows.

Reply:
It's good to see the team working so effectively together.  💞